rental/src/main/kotlin/io/axoniq/demo/bikerental/rental/query/BikeStatusProjection.kt

```diff
@@ -0,0 +1,62 @@
+package io.axoniq.demo.bikerental.rental.query
+
+import io.axoniq.demo.bikerental.coreapi.rental.*
+import org.axonframework.eventhandling.EventHandler
+import org.axonframework.queryhandling.QueryHandler
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Component
+
+@Component
+class BikeStatusProjection(
+    private val bikeStatusRepository: BikeStatusRepository
+) {
+
+    @EventHandler
+    fun on(event: BikeRegisteredEvent) {
+        bikeStatusRepository.save(BikeStatus(event.bikeId, event.bikeType, event.location))
+    }
+
+    @EventHandler
+    fun handle(event: BikeRequestedEvent) {
+        bikeStatusRepository
+            .findById(event.bikeId)
+            .map { it.requestedBy(event.renter) }
+            .ifPresent { bikeStatusRepository.save(it) }
+    }
+
+    @EventHandler
+    fun on(event: BikeInUseEvent) {
+        bikeStatusRepository
+            .findById(event.bikeId)
+            .map { it.rentedBy(event.renter) }
+            .ifPresent { bikeStatusRepository.save(it) }
+    }
+
+    @EventHandler
+    fun on(event: BikeReturnedEvent) {
+        bikeStatusRepository
+            .findById(event.bikeId)
+            .map { it.returnedAt(event.location) }
+            .ifPresent { bikeStatusRepository.save(it) }
+    }
+
+    @EventHandler
+    fun on(event: RequestRejectedEvent) {
+        bikeStatusRepository
+            .findById(event.bikeId)
+            .map { it.returnedAt(it.location) }
+            .ifPresent { bikeStatusRepository.save(it) }
```

Comment:
Not sure if this `.save()` invocation is needed? It was not present in the original solution branch, but I don't know why?